pipelines/batch/src/main/java/org/openmrs/analytics/ResourceTag.java

```diff
@@ -32,4 +34,34 @@ public class ResourceTag implements Serializable {
   Coding coding;
   String resourceId;
   Integer tagType;
+
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
+
+    if (!(other instanceof ResourceTag)) {
+      return false;
+    }
+
+    ResourceTag resourceTag = (ResourceTag) other;
+
+    return coding != null
+        && coding.equalsDeep(resourceTag.coding)
+        && StringUtils.compare(resourceId, resourceTag.getResourceId()) == 0
+        && ObjectUtils.compare(tagType, resourceTag.getTagType()) == 0;
+  }
+
+  @Override
+  public int hashCode() {
+    int h$ = 1;
+    h$ *= 1000003;
```

Comment:
nit: use `int prime = 1000003` (even define it as a constant/`static final` class attribute) and reference it here and everywhere below.

Reply:
done